sklearn/utils/estimator_checks.py

```diff
@@ -1188,7 +1194,12 @@ def check_estimators_pickle(name, estimator_orig):
 
     for method in result:
         unpickled_result = getattr(unpickled_estimator, method)(X)
-        assert_allclose_dense_sparse(result[method], unpickled_result)
+        if name == 'ChainedImputer':
+            tol = {'rtol': 1e-5, 'atol': 0.1}
```

Comment:
Is it harmful to have the same tolerance for all the estimators?

Reply:
These values are pretty high. If other estimators pass with default tolerances (`rtol=1e-07, atol=1e-9`) I think it might be better to keep it that way, otherwise some regression might not be detected..

Reply:
Ok so let me make it inline statement

```python
tol = {'rtol': 1e-5, 'atol': 0.1} if name == 'ChainedImputer' else {}
```

Reply:
I don't think we should need this at all if we put

```
    result = dict()
    for method in check_methods:
        if hasattr(estimator, method):
            result[method] = getattr(estimator, method)(X)
```
after

```
    # pickle and unpickle!
    pickled_estimator = pickle.dumps(estimator)
    if estimator.__module__.startswith('sklearn.'):
        assert_true(b"version" in pickled_estimator)
    unpickled_estimator = pickle.loads(pickled_estimator)
```

Reply:
yes, you are correct. Thanks for the helpful example below in the comments!